gprotorch/dataloader/mol_prop.py

```diff
@@ -81,94 +84,51 @@ def featurize(
-            frags = np.zeros((len(self.features), len(fragments)))
-            for i in range(len(self.features)):
-                mol = MolFromSmiles(self.features[i])
-                try:
-                    features = [fragments[d](mol) for d in fragments]
-                except:
-                    raise Exception(f"molecule {i} is not canonicalised")
-
-                frags[i, :] = features
-
-            return frags
-
-        def graphs():
-            return [
-                gm.construct_graph(smiles=i, config=graphein_config)
-                for i in self.features
-            ]
-
-        # auxiliary function to calculate bag of character representation of a molecular string
-        def bag_of_characters(selfies=False):
-            if selfies:  # convert SMILES to SELFIES
-                strings = [
-                    sf.encoder(self.features[i])
-                    for i in range(len(self.features))
-                ]
-            else:  # otherwise stick with SMILES
-                strings = self.features
-
-            # extract bag of character (boc) representation from strings
-            cv = CountVectorizer(
-                ngram_range=(1, max_ngram), analyzer="char", lowercase=False
-            )
-            return cv.fit_transform(strings).toarray()
-
         valid_representations = [
             "fingerprints",
             "fragments",
             "fragprints",
             "graphs",
             "bag_of_smiles",
             "bag_of_selfies",
+            "mqn",
         ]
 
         if representation == "fingerprints":
 
-            self.features = fingerprints()
+            self.features = fingerprints(
+                self.features, bond_radius=bond_radius, nBits=nBits
+            )
 
         elif representation == "fragments":
 
-            self.features = fragments()
+            self.features = fragments(self.features)
 
         elif representation == "fragprints":
 
             self.features = np.concatenate(
-                (fingerprints(), fragments()), axis=1
+                (
+                    fingerprints(
+                        self.features, bond_radius=bond_radius, nBits=nBits
+                    ),
+                    fragments(self.features),
+                ),
+                axis=1,
             )
 
+        elif representation == "mqn":
+            self.features = mqn_features(self.features)
+
         elif representation == "bag_of_selfies":
 
-            self.features = bag_of_characters(selfies=True)
+            self.features = bag_of_characters(self.features, selfies=True)
 
         elif representation == "bag_of_smiles":
 
-            self.features = bag_of_characters()
+            self.features = bag_of_characters(self.features)
 
         elif representation == "graphs":
-
-            self.features = graphs()
+            self.features = graphs(self.features, graphein_config)
```

Comment:
@a-r-j what's this Graphein config btw? Do users have to write their own configs to work with graphs?

Reply:
https://graphein.ai/notebooks/residue_graphs.html#Config